external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaDataWriter.scala

```diff
@@ -63,12 +63,5 @@ private[kafka010] class KafkaDataWriter(
 
   def abort(): Unit = {}
 
-  def close(): Unit = {
-    checkForErrors()
-    if (producer != null) {
-      producer.flush()
-      checkForErrors()
-      CachedKafkaProducer.close(producerParams)
-    }
-  }
+  def close(): Unit = {}
```

Comment:
This is safe; previous implementation cleans up the instance from the cache immediately so it actually helps a bit, but no big deal even we don't do it.

Reply:
is this related to adding the `close` API?

Reply:
So there's conflict; test code calls it as `close()`. If we would like to keep the code as it is, I can rename previous method as `invalidateProducer()` and leave it as it is.

Reply:
what's the life cycle of kafka producers? IIRC they were cached before, but that patch gets reverted.

Reply:
So there's no "return" in current Kafka producer cache and the cache evicts the expired producer on policy. Previously we force invalidating the Kafka producer when close() is explicitly called as callers of `close()` are temporarily using the producer (instead of running some query), and current code just let cache expire the producer on policy for all cases.

Reply:
SGTM then, if the life cycle of producers are controled by the cache manager.

Reply:
Btw, I just renamed origin `close()` method to `invalidateProducer()` to avoid the effect on Kafka side.